### ISSUE TITLE
Keytar not present in top level dependencies

### ISSUE THREAD

**Issue by sergiuilie** [zowe/zowe-cli#1314]
Hi! 

When using zowe-cli version `7.0.0-next.202202041954`, keytar is not present in `node_modules` of the project root directory and this causes some errors when using, in my case, `@zowe/imperative` ProfileInfo API. We tried with different versions to test and verify this and I can say that in one of the next versions `7.0.0-next.202112281543` keytar is present.
To install keytar in my `node_modules` root directory solved the issues I had.

node -v 16.14.0
npm -v 8.3.1

@t1m0thyj is aware of this as he guided us thoroughly to find this issue

**Comment by t1m0thyj** [comment]
Confirmed with @sergiuilie that this has been fixed 🙂

**Comment by vibhor-bc** [comment]
This Issue is again reproducible with zowe-cli version `7.0.0-next.202202171858`

@t1m0thyj is aware about the actual issue.

**Comment by t1m0thyj** [comment]
The recurrence of this issue was caused by our recent change to add [shrinkwrap](https://docs.npmjs.com/cli/v8/configuring-npm/npm-shrinkwrap-json) to the CLI package. Shrinkwrapping a package results in the dependency structure not being flattened. So if Zowe CLI is installed in another project, Keytar will live in `node_modules/@zowe/cli/keytar` instead of `node_modules/keytar`. This is an unfortunate side effect of using shrinkwrap, however due to recent supply chain attacks (e.g., [colors](https://github.com/advisories/GHSA-gh88-3pxp-6fm8), [node-ipc](https://github.com/advisories/GHSA-97m3-w2cp-4xx6)) we are going to continue using it.

In this case, worked with @vibhor-bc and @sergiuilie to migrate their project to use the Zowe SDKs. We do not shrinkwrap the SDKs, only the CLI and CLI plug-in packages, per npm's guidance that "it's strongly discouraged for library authors to publish [the shrinkwrap] file".

For future reference, here's an example of how to migrate dependencies for an app that uses the ProfileInfo and SubmitJobs APIs:

**Before**
```
  "dependencies": {
    "@zowe/cli": "6.37.8",
    "@zowe/imperative": "4.18.0"
  }
```
**After**
```
  "dependencies": {
    "@zowe/core-for-zowe-sdk": "6.37.8",
    "@zowe/imperative": "4.18.0",
    "@zowe/jobs-for-zowe-sdk": "6.37.8"
  },
  "optionalDependencies": {
    "keytar": "7.8.0"
  }
```

In some cases, migrating a project to use the Zowe SDKs may not be feasible, so we have made the following enhancements/fixes:
* Imperative is now exported as a namespace from the `@zowe/cli` package (#1347), so you can `import { imperative } from "@zowe/cli"` to use the ProfileInfo API even if Imperative is not a top-level dependency in your project's node_modules.
* If you import Imperative in this way, it should find Keytar in the node_modules folder alongside Imperative. We fixed a bug (zowe/imperative#760) where the ProfileInfo API would only find Keytar in the top-level node_modules folder.
